src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -91,49 +91,16 @@ public without sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGa
      */
     public static list<Trigger_Handler__c> listTH {
         get {
-            if (listTH == null) {
-                listTH =  [select Asynchronous__c, Class__c, Load_Order__c, Object__c, Active__c, Usernames_to_Exclude__c, 
-                    Trigger_Action__c from Trigger_Handler__c order by Load_Order__c];
+            if(ADV_Info.useAdv() && ADV_Info.getApiNPSP() != null) {
+                listTH = ADV_Info.getApiNPSP().getDefaultTdtmConfigNpspRecords(listTH);
```

Comment:
so by having Adv always reset listTH, it is preventing any customer code from modifying listTH in the transaction, which NPSP says it supports in TDTM_Config_API.getCachedRecords().

    * @description Returns the list of current cached trigger handler records
    * @return List<Trigger_Handler__c> a list of cached trigger handler records with current values.  SI's
    * may modify this list in the current execution context to temporarily change the functionality of
    * tdtm triggers - disable triggers, enable triggers, or even add a new row to the collection
    * that is not committed to the database.

is this intentional?  if so, I think we need an explicit comment in the code.

@ceiroa want to make sure you see this discussion.

Reply:
Actually, no. Adv will not totally reset the listTH. What Adv will do, it to take whatever in the listTH and disable some TriggerHandlers that will conflict with HEDA in order to make sure Adv works correctly. 

1. If customer only have NPSP, then they can do whatever they want since the logic is still the same.

2. If customer have NPSP + Adv, it will not prevent the customer call TDTM_Config_API.getCachedRecords(). Adv will only disable the TriggerHandler that has to be disable and then return the rest back. 

3. If later anything need to be update, we can just update it in Adv package and leave NPSP untouch since the logic is in Adv

I will double check the comments to make sure it makes sense.

Reply:
Right, at the end the call goes to https://github.com/SalesforceFoundation/adv/pull/2/files#diff-3ea107cf2322681898732395ee4951f2R92, which just deactivates some NPSP TMTD classes. That looks correct to me...